Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -279,21 +305,65 @@ def get_diy_account_id():
     except:
         return None
 
+def get_optional_metadata():
+    unknown = "Unknown"
+    asset_tag = unknown
+    vm_id = unknown
+    is_azure_vm = False
+    try:
+        proc = subprocess.Popen(["sudo", "-u", AUTOMATION_USER, "python", OMS_UTIL_FILE_PATH, "--dmidecode"],
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        dmidecode, error = proc.communicate()
+        if proc.returncode != 0 or not dmidecode:
+            raise Exception("Unable to invoke omsutil.py --dmidecode: %s" % error)
+        is_azure_vm = linuxutil.is_azure_vm(dmidecode)
+        if is_azure_vm:
+            asset_tag = linuxutil.get_azure_vm_asset_tag()
+        vm_id = linuxutil.get_vm_unique_id_from_dmidecode(sys.byteorder, dmidecode)
+    except Exception, e:
+        log(INFO, "unable to get_optional_metadata: %s" % str(e))
+
+    return asset_tag, is_azure_vm, vm_id
+
 
 def get_manually_registered_worker_conf_path(workspace_id):
     return "/var/opt/microsoft/omsagent/%s/state/automationworker/diy/worker.conf" % workspace_id
 
 
+def is_certificate_valid(worker_conf_path, certificate_path):
+    """
+    certificate is vaild when thumbprints match
```

Comment:
General comment, If possible try to mention what the method does in the comment "Validates that certificated thumbprint are matching" then you can add more detailed explenation. Also try to have similar format for you method comments.